src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerFileReferenceTests.cs

```diff
@@ -597,4 +644,25 @@ public class VisualBasicAndCSharpGenerator : TestGenerator { }
 
     [Generator(LanguageNames.FSharp)]
     public class FSharpGenerator : TestGenerator { }
+
+    [Generator(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public class TestIncrementalGenerator : IIncrementalGenerator
+    {
+        public void Initialize(IncrementalGeneratorInitializationContext context) => throw new NotImplementedException();
+    }
+
+    public class TestIncrementalGeneratorWithNoAttrib : IIncrementalGenerator
+    {
+        public void Initialize(IncrementalGeneratorInitializationContext context) => throw new NotImplementedException();
+    }
+
+    [Generator(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public class TestSourceAndIncrementalGenerator : IIncrementalGenerator, ISourceGenerator
```

Comment:
just curious: is this a supported scenario? or is this something we should discourage people from doing via an analyzer?

Reply:
I think there is value in it being supported, particularly while the feature is being previewed. Lets authors sim ship their generators in both models and it continues working when the developer flips the preview flag but also leverages the new model